src/Microsoft.AspNetCore.SignalR.Redis/Internal/AckHandler.cs

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Concurrent;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.AspNetCore.SignalR.Redis.Internal
+{
+    internal class AckHandler : IDisposable
+    {
+        private readonly ConcurrentDictionary<int, AckInfo> _acks = new ConcurrentDictionary<int, AckInfo>();
+        private readonly Timer _timer;
+        private readonly TimeSpan _ackThreshold = TimeSpan.FromSeconds(30);
+        private readonly TimeSpan _ackInterval = TimeSpan.FromSeconds(5);
+        private readonly object _lock = new object();
+        private bool _disposed;
+
+        public AckHandler()
+        {
+            _timer = new Timer(_ => CheckAcks(), state: null, dueTime: _ackInterval, period: _ackInterval);
+        }
+
+        public Task CreateAck(int id)
+        {
+            lock (_lock)
+            {
+                if (_disposed)
+                {
+                    return Task.CompletedTask;
```

Comment:
I think this causes issues - you will return a completed task and will still try to send a message over Redis (which is probably disposed as well)...

Reply:
Thats fine, you'll just get whatever error you normally would have gotten when trying to publish a message from a disposed redis instance. This doesn't change anything

Reply:
I was wondering if we should not have a `_disposed` flag in the RedisLifetimeManager and throw/not handle requests after it was disposed (not really sure how likely it is to happen)

Reply:
We might want to consider this for preview, but don't think it is related to this change.

Reply:
Sound good.